core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -290,8 +291,13 @@ class AdminManager(val config: KafkaConfig,
           throw new InvalidPartitionsException(s"Topic already has $oldNumPartitions partitions.")
         }
 
-        val newPartitionsAssignment = Option(newPartition.newAssignments).map(_.asScala.map(_.asScala.map(_.toInt))).map { assignments =>
-          val unknownBrokers = assignments.flatten.toSet -- allBrokerIds
+        val newPartitionsAssignment = Option(newPartition.assignments()).map(_.asScala)
```

Comment:
nit: this copy to change to scala types is a little annoying. Couldn't we do this in the first pass below?

Reply:
Done. Its a bit cleaner now.